evm-node.md

```diff
@@ -0,0 +1,338 @@
+EVM Integration with Production Client
+======================================
+
+Contained in this file is glue code needed in order to enable the ability to use KEVM as a VM for an actual Ethereum node.
+
+```{.k .node}
+requires "evm.md"
+requires "optimizations.md"
+
+module EVM-NODE
+    imports EVM
+    imports EVM-OPTIMIZATIONS
+    imports K-REFLECTION
+    imports COLLECTIONS
+    imports BYTES
+```
+
+### State loading operations.
+
+In order to enable scalable execution of transactions on an entire blockchain, it is necessary to avoid serializing/deserializing the entire state of all accounts when constructing the initial configuration for KEVM.
+To do this, we assume that accounts not present in the `<accounts>` cell might not exist and need to be loaded on each access.
+We also defer loading of storage entries and the actual code byte string until it is needed.
+Because the same account may be loaded more than once, implementations of this interface are expected to cache the actual query to the Ethereum client.
+
+-   `#unloaded` represents the code of an account that has not had its code loaded yet. Unloaded code may not be empty.
+-   Empty code is detected without lazy evaluation by means of checking the code hash, and therefore will always be represented in the `<code>` cell as `.WordStack`.
+
+```{.k .node}
+    syntax InternalOp ::= "#load" "[" OpCode "]"
+ // --------------------------------------------
+    rule <k> #next [ OP ]
+          => #if isAddr1Op(OP) orBool isAddr2Op(OP) #then #addr [ OP ] #else . #fi
+          ~> #load [ OP ]
+          ~> #exec [ OP ]
+          ~> #pc   [ OP ]
+         ...
+         </k>
+         <wordStack> WS </wordStack>
+         <static> STATIC:Bool </static>
+      requires notBool ( #stackUnderflow(WS, OP) orBool #stackOverflow(WS, OP) )
+       andBool notBool ( STATIC andBool #changesState(OP, WS) )
+    [priority(35)]
```

Comment:
We don't want to override the entire `#next` rule, because it will obliterate the existing optimizations. The other drawback is that we then have to keep this in sync.

Instead, we should do something like:

```
configuration <kevm-node> <kevm/> <loaded> false </loaded> </kevm-node>

rule
<k> (. => #load [ OP ]) ~> #next [ OP ] ... </k>
<loaded> false => true </loaded>
requires isAddr1Op(OP) orBool isAddr2Op(OP) orBool OP ==K CREATE orBool OP ==K SSTORE orBool OP ==K SLOAD
 [priority(35)]

rule <k> #execute ... </k> <loaded> true => false </loaded> [priority(35)]
```